PencilCase/PencilCase/scripts/Pages/ImportProduct.ts

```diff
@@ -144,17 +144,14 @@ export class ImportProduct extends PageBase {
             let newRetailCost = ((product.RetailCost * product.Inventory) + (order.price() * order.quantity())) / (product.Inventory + order.quantity() * product.Times);
             let newWholesaleCost = newRetailCost * product.Times;
 
-            product.Inventory += (order.quantity() * product.Times);
-            product.RetailCost = newRetailCost;
-            product.WholesaleCost = newWholesaleCost;
-            product.ImportWholesalePrice = order.price();
-            product.ImportRetailPrice = product.ImportWholesalePrice / product.Times;
-            product.CreatedDate = new Date(product.CreatedDate);
-            product.ModifiedDate = new Date(Date.now());
-
             this.orderRepository.insert(order, (transaction: SqlTransaction, resultSet: SqlResultSet) => {
-                this.productRepository.update(product, (transaction: SqlTransaction, resultSet: SqlResultSet) => {
-                    
+                this.productRepository.updateWithFieldValues([
+                    { Field: "Inventory", Type: "number", Value: "Inventory + " + (order.quantity() * product.Times) },
```

Comment:
This is an informal, special way to update the inventory value. In this way, it can make sure the Inventory used to do calculation always be right.